nixos/modules/config/pulseaudio.nix

```diff
@@ -9,11 +9,33 @@ let
 
   systemWide = cfg.enable && cfg.systemWide;
   nonSystemWide = cfg.enable && !cfg.systemWide;
+  hasZeroconf = let z = cfg.zeroconf; in z.publish.enable || z.discovery.enable;
+
+  overriddenPackage = cfg.package.override
+    (optionalAttrs hasZeroconf { zeroconfSupport = true; });
+  binary = "${getBin overriddenPackage}/bin/pulseaudio";
+  binaryNoDaemon = "${binary} --daemonize=no";
 
   # Forces 32bit pulseaudio and alsaPlugins to be built/supported for apps
   # using 32bit alsa on 64bit linux.
   enable32BitAlsaPlugins = cfg.support32Bit && stdenv.isx86_64 && (pkgs_i686.alsaLib != null && pkgs_i686.libpulseaudio != null);
 
+
+  myConfigFile =
+    let
+      addModuleIf = cond: mod: optionalString cond ''
+        echo "load-module ${mod}" >> "$out"
+      '';
+      allAnon = optional cfg.tcp.anonymousClients.allowAll "auth-anonymous=1";
+      ipAnon =  let a = cfg.tcp.anonymousClients.allowedIpRanges;
+                in optional (a != []) ''auth-ip-acl=${concatStringsSep ";" a}'';
+    in runCommand "default.pa" {} ''
+      cat "${cfg.configFile}" > "$out"
```

Comment:
This changes the semantics of the `configFile` option. Previously, this was a path to the configuration file on the host (like `/etc/pulse/default.pa`) to be. Now this has to be a file that is available at evaluation time on the evaluating machine.

I for one welcome this change but this should probably be documented.